partition: add doc for global index, also with GLOBAL IndexOption/ColumnOption (#18543)
pingcap/docs: sql-statements/sql-statement-create-index.md

* Expression indexes are incompatible with views. When a query is executed using a view, the expression index cannot be used at the same time.
* Expression indexes have compatibility issues with bindings. When the expression of an expression index has a constant, the binding created for the corresponding query expands its scope. For example, suppose that the expression in the expression index is `a+1`, and the corresponding query condition is `a+1 > 2`. In this case, the created binding is `a+? > ?`, which means that the query with the condition such as `a+2 > 2` is also forced to use the expression index and results in a poor execution plan. In addition, this also affects the baseline capturing and baseline evolution in SQL Plan Management (SPM).
* The data written with multi-valued indexes must exactly match the defined data type. Otherwise, data writes fail. For details, see [create multi-valued indexes](/sql-statements/sql-statement-create-index.md#create-multi-valued-indexes).

> * Creating an `UNIQUE KEY` with `GLOBAL` IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.

Reviewer: ```suggestion
* Setting a `UNIQUE KEY` as a global index with `GLOBAL` IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.
```

> * Creating an `UNIQUE` `KEY` with 'GLOBAL' IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.

Reviewer: ```suggestion
* Creating an `UNIQUE KEY` with 'GLOBAL' IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.
```

> * Creating an `UNIQUE` `KEY` with 'GLOBAL' IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.

Reviewer: Should `UNIQUE` `KEY` be `UNIQUE KEY`? In addition, it appears in the [create table](https://github.com/pingcap/docs/pull/18543/files#diff-05d7fa31873281b59e53a89cec973d1c909ee9fe91fde46ce677bb4700cd4cd7L49) doc, instead of this doc.

Author: I don't have any strong preference, I had them as separate to distinguish from `PRIMARY KEY` :)
What do you prefer?

Author: I updated it to `UNIQUE KEY`.

> * Creating an `UNIQUE` `KEY` with 'GLOBAL' IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.

Author: ```suggestion
* Creating an `UNIQUE KEY` with `GLOBAL` IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.
```

> * Setting a `UNIQUE KEY` as a global index with `GLOBAL` IndexOption is a TiDB extension for [global indexes](/partitioned-table.md#global-indexes) and is not compatible with MySQL.

Reviewer 2: ```suggestion
* Setting a `UNIQUE KEY` as a [global index](/partitioned-table.md#global-indexes) with the `GLOBAL` index option is a TiDB extension for [partitioned tables](/partitioned-table.md) and is not compatible with MySQL.
```
